### PR TITLE
Initial commit of release automation tooling.

### PR DESCRIPTION
**Pull request by david-mcmahon**
There are many TODOs, but this is a good working baseline from which to build.
We're producing releases using this tooling.

### CODE REVIEW

**Comment by david-mcmahon** [comment]
cc @mikedanese @ihmccreery @eparis @bgrant0607 @fgrzadkowski @zmerlynn

**Comment by ihmccreery** [comment]
@david-mcmahon Do you want a review of this, or not yet?

**Comment by david-mcmahon** [comment]
@ihmccreery I'm continuing to add to it, but feel free to comment.

**Comment by david-mcmahon** [comment]
ref https://github.com/kubernetes/kubernetes/issues/19312

**Comment by ihmccreery** [comment]
See https://github.com/kubernetes/kubernetes/issues/24535; not sure if it's related, but certainly seems possible, if you cut `alpha.1` from this tool.

**Comment by ihmccreery** [comment]
Also, it looks like the `latest.txt` files aren't getting updated:

```
$ gsutil cat gs://kubernetes-release/release/latest.txt
v1.3.0-alpha.1
```

**Comment by ihmccreery** [comment]
Let me know if you'd like me to file an issue about that as well.

**Comment by david-mcmahon** [comment]
Yes, thanks for the catch.  Commented in kubernetes/kubernetes#24535.

**Comment by david-mcmahon** [comment]
This has been a WIP for too long.  'Anago' has been doing releases now for some months and this repo is starting to get more attention/references from other subrepos.
It is XXL, but all of the pieces are inter-related for this initial commit.

Can someone give this a look?

cc @ihmccreery @eparis @bgrant0607 @fgrzadkowski @zmerlynn @thockin

**Comment by bgrant0607** [comment]
I'm going to merge this and we can iterate from there.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,107 @@
...
+
+```
+$ anago -man
+```
+
+The idea is that no external doc updates should be necessary and the
+tool itself contains all of the details and instructions and prerequisite
+checks needed for anyone to run the tool in at least mock mode.
+
+There is a simple $USER check to ensure that noone but a certain few people can
```
Comment (ihmccreery) [inline README.md]: no one

File: README.md [hunk 2]
```diff
@@ -1,2 +1,107 @@
...
+run the script with --nomock to perform a real release.
```
Comment (ihmccreery) [inline README.md]: `--nomock`

File: README.md (hunk 1 above)
Comment (ihmccreery) [inline README.md]: `$USER`

**Pull request merged**